cql3/statements/describe_statement.cc

```diff
@@ -142,6 +144,82 @@ future<std::vector<description>> generate_descriptions(
     co_return descs;
 }
 
+template <std::ranges::range Range>
+    requires std::is_base_of_v<keyspace_element, std::ranges::range_value_t<Range>>
+future<std::vector<description>> generate_descriptions(replica::database& db, const Range& elements,
+    std::optional<bool> with_internals = std::nullopt, bool sort_by_name = true)
+{
+    std::vector<description> descs{};
+
+    // In general, a range is allowed to be modified during the call to `std::ranges::size()`.
+    // However, we rely on the sanity of the passed types and the fact that we take the range
+    // as a const-reference.
+    descs.reserve(std::ranges::size(elements));
+
+    for (auto& e: elements) {
+        auto desc = (with_internals.has_value())
+            ? description(db, e, *with_internals)
+            : description(db, e);
+        descs.push_back(std::move(desc));
+
+        co_await coroutine::maybe_yield();
+    }
+
+    if (sort_by_name) {
+        std::ranges::sort(descs, [] (const auto& lhs, const auto& rhs) {
```

Comment:
You can write std::ranges::sort(descs, std::less<>, std::mem_fn(&description::_name)) which would save me the trouble of complaining about the auto parameters.